File: 13-axios-and-react-router/backend/src/main/java/ru/sber/services/PaymentService.java
package ru.sber.services;

import ru.sber.entities.Payment;

/**
 * Сервис для оплаты товаров
 */
public interface PaymentService {

    /**
     * Совершает платеж
     *
     * @param payment Платеж
     * @return Возвращает статус выполнения платежа
     */
    boolean pay(Payment payment);

}

File: 14-spring-security/frontend/src/services/paymentService.js
import axios from "axios";
import UserService from "./userService";
import authHeader from "./auth-header";
import CartService from "./cartService";

const API_URL = "http://localhost:8081/payment";

const pay = (payment, dispatch) => {

    return axios.post(API_URL, payment, {headers: authHeader()}).then(
        () => {
            UserService.getUser(payment.userId, dispatch)
            CartService.clearCart(payment.userId, dispatch)
        },
        (error) => {
            const _content = (error.response && error.response.data) ||
                error.message ||
                error.toString();

            console.error(_content)
        });
};

const paymentService = {
    pay,
};

export default paymentService